services/venice-router/src/main/java/com/linkedin/venice/router/stats/SecurityStats.java

```diff
@@ -46,10 +41,10 @@ public void recordNonSslRequest() {
   }
 
   /**
-   * This function will be triggered in every ssl event.
+   * This function will be triggered in {@link com.linkedin.alpini.netty4.handlers.ConnectionControlHandler}.
    */
-  private void recordLiveConnectionCount() {
-    this.sslLiveConnectionCount.record(secureConnectionCountSupplier.getAsInt());
+  public void recordLiveConnectionCount(int connectionCount) {
```

Comment:
The end result might not be as what we expected..
For example, if there is no new connections within 2 time windows, the value of the metric will be `NaN`.
One way is to maintain an integer field here, so that `ConnectionControllerHandler` or `ConnectionLimitHandler` can update it when creating new conns or closing any existing conns, and this method will be triggered by any new request.

That means, when there is no traffic, this metric would report `NaN`, which might be acceptable, or we can do min/max/avg/gauge, so when there are traffic, `min/max/avg` will be more meaningful, and when there is no traffic, `gauge` will give us per-min data point.

Reply:
Oh, good point. Do you mean the below implementation:
1. Maintain two counters, one for avg and the other for max; ConnectionLimitHander would update these two counters
2. Register two gauge metrics which get metric values from these two counters?

Reply:
Let us sync up offline on ^.